mne/io/base.py

```diff
@@ -212,14 +212,56 @@ def _check_fun(fun, d, *args, **kwargs):
 
 class _BaseRaw(ProjMixin, ContainsMixin, PickDropChannelsMixin,
                SetChannelsMixin, InterpolationMixin, ToDataFrameMixin):
-    """Base class for Raw data"""
+    """Base class for Raw data
+
+    Subclasses must provide the following methods:
+
+        * _read_segment(start, stop, sel, projector, verbose)
+          (only needed for types that support on-demand disk reads)
+    """
     @verbose
-    def __init__(self, *args, **kwargs):
-        raise NotImplementedError
+    def __init__(self, info, data=None, cals=None,
+                 first_samps=(0,), last_samps=None,
+                 filenames=(), rawdirs=(),
+                 comp=None, orig_comp_grade=None,
+                 orig_format='double', verbose=None):
+        self.info = info
+        self._data = data
+        if cals is None:
+            cals = np.empty(info['nchan'])
+            for k in range(info['nchan']):
+                cals[k] = info['chs'][k]['range'] * info['chs'][k]['cal']
```

Comment:
there is an inconsistency in how `Raw.cals` is defined across modules. For some of the eeg modules, 
`RawEEG.cals = [ch['cal'] for ch in RawEEG.info['chs']]`
whereas 
`Raw.cals = [ch['cal']*ch['range'] for ch in Raw.info['chs']]`.

when `range` != 1, this can be a problem.